tutorials/functional-full-stack/purescript-bridge/tutorial.md

```diff
@@ -0,0 +1,384 @@
+---
+title: Connecting a Haskell Backend to a PureScript Frontend
+published: not-yet
+ghc: 7.10.3
+purs: 0.10.5
+lts: 7.17
+libraries: purescript-bridge
+language: haskell purescript
+author-name: Javier Casas Velasco
+description: In this tutorial we will implement a way to extend the types in the Haskell backend to the PureScript frontend while maintaining consistency and simplifying communication.
+---
+# Connecting a Haskell Backend to a PureScript Frontend
+## Introduction
+At [Stack Builders](https://www.stackbuilders.com) we are working on a full-stack app with [CollegeVine](https://www.collegevine.com/) using Functional Languages.
+We have a [Haskell](https://www.haskell.org) backend written in [Servant](http://haskell-servant.readthedocs.io) that manipulates the database and offers some endpoints to a [PureScript](http://www.purescript.org) frontend,
+that does all the React-like magic to show a really nice interface on the user's browser.
+It's great because we have advanced types, purity and all the awesome benefits that offers the Functional World.
+But not everything is perfect.
+
+### Motivation
+The problem is that we have two different codebases: one in Haskell and the other in PureScript.
+The syntax is almost the same, but not exactly the same. For example, on Haskell we do:
+```Haskell
+{-# LANGUAGE DeriveGeneric #-}
+
+data Blah = Blah
+  {
+    bFoos :: [Foo]
+  } deriving (Generic)
+```
+Whereas if I want to get a Generic instance for my `Blah` type, in PureScript I have to do:
```

Comment:
@javcasas Usually, tutorials are written using third person form. What do you think about rewriting this paragraph as follows:

```
Whereas, if we want to get a Generic instance for `Blah` type, in PureScript we have to do:
```